src/components/atoms/FloatButtonNav/index.tsx

```diff
@@ -1,31 +1,40 @@
 "use client";
 
 import { StaticImageData } from "next/image";
-import React from "react";
+import React, { useState } from "react";
 import Image from "next/image";
 import NextLink, { LinkProps } from "next/link";
 import { UrlObject } from "url";
 import { motion } from "framer-motion";
+import LoadingComponent from "../Loading";
 
 interface IFloatButtonNavProps {
   icon: StaticImageData;
   backgroundStyle: "main" | "secondary";
+  label?: string;
   href: UrlObject | string;
   hasCouponActive?: boolean;
 }
 
 const FloatButtonNav: React.FC<IFloatButtonNavProps & LinkProps<any>> = ({
   href,
+  label,
   backgroundStyle,
   icon,
   hasCouponActive = false,
 }) => {
+  const [stateLabel, setStateLabel] = useState<any>(label);
+
+  const onClick = () => {
+    setStateLabel(<LoadingComponent bgColor="none" fullSize={false} />);
+  };
+
```

Comment:
Let's improve this:

- What's "state label"? It doesn't mean anything clear to me reading the first time. 
- Why are we adding a component to the state? Why aren't we using a flag and rendering the component based on this?